src/librbd/WatchNotifyTypes.cc

```diff
@@ -208,16 +208,22 @@ void AsyncCompletePayload::dump(Formatter *f) const {
 
 void ResizePayload::encode(bufferlist &bl) const {
   ::encode(size, bl);
+  ::encode(allow_shrink, bl);
   AsyncRequestPayloadBase::encode(bl);
 }
 
 void ResizePayload::decode(__u8 version, bufferlist::iterator &iter) {
   ::decode(size, iter);
+
```

Comment:
Since this is a protocol change, it actually needs to be the last thing decoded. Therefore, move to after ```AsyncRequestPayloadBase::decode``